### PR TITLE
Tsc 433 wrap rotate text in columns

### PR DESCRIPTION
**Pull request by JacquiFreeElectron**
all Zone columns will have "orientation" radio group. the default value should be "normal", which is shown as "Horizontal" to users. the only exception is "Series Label", it should have "vertical" as its default orientation.

demo:

https://github.com/user-attachments/assets/164c0faf-5397-440d-9709-b4cd7002b6ec

block-Zone
chron label-Zone
facies label-Zone
members-Zone
series label-Zone
another sereis label-Zone

### CODE REVIEW

File: app/src/settings_tabs/column_menu/ColumnMenu.tsx [hunk 1]
```diff
@@ -170,6 +183,17 @@ const ColumnContent: React.FC<ColumnContentProps> = observer(({ tab, column }) =
...
+              <TSCRadioGroup
+                onChange={changeZoneColumnOrientation}
+                name={"Label Orientation"}
+                value={isZoneColumn()}
+                radioArray={[
+                  { value: "normal", label: "Horizontal" },
+                  { value: "vertical", label: "Vertical" }
+                ]}
+              />
+            )}
```
Comment (paologum) [inline app/src/settings_tabs/column_menu/ColumnMenu.tsx:196]: there are more block specific settings, so you should make a component in `advanced_settings` like `EventSpecificSettings` and `RangeSpecificSettings`,

additionally, this should be above the info box.
idk if i forgot to mention but the ruler age column above this should also be above the info box

File: app/src/settings_tabs/column_menu/ColumnMenu.tsx [hunk 2]
```diff
@@ -116,6 +120,15 @@ const ColumnContent: React.FC<ColumnContentProps> = observer(({ tab, column }) =
...
   };
+  const isZoneColumn = () => {
+    if (column.columnDisplayType !== "Zone") return "";
+    try {
+      assertZoneSettings(column.columnSpecificSettings);
+      return column.columnSpecificSettings.orientation;
+    } catch (e) {
+      return "";
+    }
+  };
```
Comment (paologum) [inline app/src/settings_tabs/column_menu/ColumnMenu.tsx:131]: when you implement it in the component have it just return null

File: app/src/state/parse-settings.ts [hunk 3]
```diff
@@ -396,7 +397,16 @@ export function translateColumnInfoToColumnInfoTSC(state: ColumnInfo): ColumnInf
...
-      column = cloneDeep(defaultZoneColumnInfoTSC);
+      if (state.name != "Chart Root") {
+        assertZoneSettings(state.columnSpecificSettings);
+        column = {
+          ...cloneDeep(defaultZoneColumnInfoTSC),
+          orientation: state.columnSpecificSettings.orientation
+        };
+      } else {
+        column = cloneDeep(defaultZoneColumnInfoTSC);
+      }
```
Comment (paologum) [inline app/src/state/parse-settings.ts:408]: i'm confused, `Chart Root` is a `RootColumn` not a `ZoneColumn`

**Review by paologum — CHANGES_REQUESTED** [review]
just some small things but looks great so far

File: app/src/state/parse-settings.ts (hunk 3 above)
Comment (JacquiFreeElectron) [reply app/src/state/parse-settings.ts:408]: idk, one time when I ran `yarn test` it showed there's a Chart Root `ZoneColumn` and threw an error, and after I added the `if (state.name != "Chart Root") {...}` the error dissappeared. I can't remember what exactly the error was tho. I'll double check

Comment (JacquiFreeElectron) [reply app/src/state/parse-settings.ts:408]: so there's a test has a chart root column with display type Zone, but the real chart root column doesn't have display type `Zone`. Do you want me to change that test?

Comment (paologum) [reply app/src/state/parse-settings.ts:408]: the problem should then be that the test needs to have a valid `ZoneSettings` in `columnSpecificSettings`, not to ignore it so yes

Comment (paologum) [inline app/src/state/parse-settings.ts:408]: you should revert this to not exclude `Chart Root`

File: shared/src/index.ts [hunk 4]
```diff
@@ -633,6 +638,14 @@ export function assertRulerSettings(o: any): asserts o is RulerSettings {
...
     throwError("RulerSettings", "justification", "string and left | right", o.justification);
 }
+export function isZoneOrientation(o: any): o is ZoneOrientation {
+  return /^(normal|vertical)$/.test(o);
+}
+export function assertZoneSettings(o: any): asserts o is ZoneSettings {
+  if (!o || typeof o !== "object") throw new Error("ZoneSettings must be a non-null object");
+  if (typeof o.orientation !== "string" && isZoneOrientation(o.justification))
+    throwError("ZoneSettings", "orientation", "string and normal | vertical", o.orientation);
+}
```
Comment (paologum) [inline shared/src/index.ts:648]: `!isZoneOrientation`

**Review by paologum — CHANGES_REQUESTED** [review]

File: app/src/state/parse-settings.ts (hunk 3 above)
Comment (JacquiFreeElectron) [reply app/src/state/parse-settings.ts:408]: oops I forgot

**Comment by paologum** [comment]
the deployment has errors when i select a column

**Comment by JacquiFreeElectron** [comment]
> the deployment has errors when i select a column

I mistakenly put `assertRulerSettings` before check if it is a Ruler column

**Review by paologum — APPROVED** [review]

**Pull request merged**